src/components/button/button.tsx

```diff
@@ -1,43 +1,87 @@
 import { ReactNode } from 'react';
 import classNames from 'classnames';
 import styles from './button.module.scss';
+import { BreakpointKeys, breakpoints, useWindowDimensions } from '../../hooks/use-window-dimensions';
 
-export type Props = {
+export type Size = 'xxs' | 'xs' | 'sm' | 'md' | 'lg' | 'normal' | 'large'; //TODO: remove normal and large after rebranding is done;
+
+type BreakpointsProps = { [key in BreakpointKeys]?: { size?: Size } };
+
+export interface Props extends BreakpointsProps {
   children: ReactNode;
   className?: string;
-  type?: 'primary' | 'secondary' | 'link' | 'text';
-  size?: 'normal' | 'large';
+  type?: 'primary' | 'secondary' | 'link' | 'text' | 'menu-link-secondary' | 'link-blue';
```

Comment:
Hmm I think we are going down a weird path here. The "link" variant for the Button component was to give us an actual button element that has similar styling to a link. I understand a Button component that is more flexible in that it can also render an anchor element that has similar styling to a button.

But now it seems this Button component can be used to give us an anchor element that looks like a link:
```tsx
<Button type="link|link-blue|menu-link-secondary" href="..." ... />
```
^^ This usage does not make sense, as it neither renders a button, or styles the anchor as a button.

I think this `link-blue` and `menu-link-secondary` should be removed here. We've got an `<ExternalLink />` component which should probably be extended.

Reply:
Yeah, I can see your point here 🤔 My approach is rooted in [the design library](https://www.figma.com/design/BhnOcskTIGPZ18JkhbCuNQ/%E2%9F%81--Api3-DS---Branding-Update?node-id=322-26668&node-type=section&m=dev), where we have both links and buttons categorized simply as buttons. 

I suppose we could have `link-blue` and `menu-link-secondary` as part of the `<ExternalLink />` as you suggested, but my worries are that:
- The `<ExternalLink />` would share lots of similar traits with the `<Button />` - props like `type, theme, size, disabled, href`. Having two components with almost the same `props` would feel repetitive. 
- It would complicate the code in situation like the footer where we have both 
  - links to external sites ("About API3", "GitHub") and 
  - button opening the Error reporting bar ("Error reporting"). 
- All of these ^^ footer "elements" should be rendered the same while being both buttons and links. If you went through designs you would find more situations like this.

If you feel that component that could be both link and the button shouldn't be actually named "Button", maybe we can come up with something more neutral? I feel like this is more about our own organization of the code than the actual HTML output, since in both approaches the result is: an anchor element for links and button element for buttons.

Reply:
Ok, I don't want to spend too much time arguing about this, given we have quite a tight deadline now. So let's move to the solution. What do you propose? To remove `link-blue` and `menu-link-secondary` from the `<Button />` and make it part of the `<ExternalLink />` component? How would you implement buttons that have `link-blue` and `menu-link-secondary` style in the designs? There are couple of them - aforementioned "Error Reporting" in the footer and then these `link-blue` ones:

![image](https://github.com/user-attachments/assets/01740aaa-8480-43b9-98b6-e60b88bb0a3c)

![image](https://github.com/user-attachments/assets/0d40fa1b-85c2-425c-974f-4e848a51ed7e)

![image](https://github.com/user-attachments/assets/742ec915-7768-44c2-8340-819e4da0eb58)

![image](https://github.com/user-attachments/assets/7871403b-823c-4621-9468-7af141cbb1a6)


If you just don't want to have `<Button />` component rendering as `<a>`  we can use `@mixins` to define `link-blue` style somewhere and reuse it in both `<Button type="link-blue" />` and  `<ExternalLink type="link-blue" />` and we would use ExternalLink in the case it's an external link and button in every other case. Does it work for you?

Reply:
> If you just don't want to have <Button /> component rendering as <a> we can use @mixins to define link-blue style somewhere and reuse it in both <Button type="link-blue" /> and <ExternalLink type="link-blue" /> and we would use ExternalLink in the case it's an external link and button in every other case. Does it work for you?

Appreciate the detailed response. Yeah this ^ sounds like a good option, I'm happy to go with it 👍

Reply:
Done ✅